plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java

```diff
@@ -23,10 +23,10 @@
 
 import static java.lang.String.format;
 
-public class TestElasticsearch6IntegrationSmokeTest
-        extends BaseElasticsearchSmokeTest
+public class TestElasticsearch6ConnectorTest
```

Comment:
@hashhar one question here. You suggested that `BaseConnectorTest` is used for testing ES6 while `BaseConnectorSmokeTest` for ES7. Not sure if that matters but maybe we should test newest version with more verbose `BaseConnectorTest`?

Reply:
I suggested based on how Cassandra/MemSQL/Mongo are tested today. I think the older ones require deeper testing since the upstream projects generally don't break things in newer versions (while something working in new version doesn't guarantee it worked in older version).

I don't have a strong opinion - very less information to make an informed decision.

Reply:
Elasticsearch 7 has been around for quite some time now  (more than 2 years actually)

https://github.com/elastic/elasticsearch/releases

Elasticsearch 6 has the latest bugfix release made about 5 months ago.
Elasticsearch 7 has the latest minor release made a month ago.

I tend to say that Elasticsearch 7 is more relevant at the time of this writing than Elasticsearch 6.

Reply:
> Elasticsearch 7 has been around for quite some time now (more than 2 years actually)
> 
> https://github.com/elastic/elasticsearch/releases
> 
> Elasticsearch 6 has the latest bugfix release made about 5 months ago.
> Elasticsearch 7 has the latest minor release made a month ago.
> 
> I tend to say that Elasticsearch 7 is more relevant at the time of this writing than Elasticsearch 6.

I would go with testing more the version is more broadly used. I have now data on that at all. But from what I read above it seems like ES7 is the one to have broader test coverage. @findepi @martint opinion?

Reply:
@losipiuk see discussion under https://github.com/trinodb/trino/issues/5804. Does this help?

Reply:
https://github.com/trinodb/trino/issues/5804#issuecomment-721365549

Maybe Elasticsearch 6 & Elasticsearch 7 full tested and a Elasticsearch 7 latest smoke test ? 

What do you think about this?

Reply:
It is safe for sure. But unless there are significant differences between 6 and 7. And both are commonly used I would rather have just one full tests, to keep CI runs as short as possible.

Reply:
Good. Then I'll keep things as they are now.

I'll just drop then the base class `BaseElasticsearchConnectorTest` and include everything in `TestElasticsearch6ConnectorTest.java ` .

Reply:
There are some significant differences between 6 and 7. Major versions of ES tend to make breaking changes in their APIs. That’s why we even added tests for both at all.

Reply:
Thanks for chiming in. Still smoke test should be enough I think. The concept of smoke test class is that it still provides good test coverage; each important code path should be tested; we are just passing on testing every tiny corner case for each code path. So IMO it is reasonable trade-off to have just one full test class, and gain on faster CI trunaround.

cc: @findepi

Reply:
The issue is that in this change we've trade off testing ES-specific behaviors for generic connector behaviors in the ES 7 smoke test. I don't think we should do that -- those tests were there for a reason, as ES 6 and 7 have differences in how they expose index mappings (the shape of the URL and contents). 

I'm ok with not running the full suite of queries, but the tests that currently exist exercise certain areas that are not covered by any of the generic smoke tests, and we should remove them (e.g, nested documents, all supported ES data types, coercions from ES types, pushdown on specific types, index aliases, etc)